medic/cht-core
# add script to archive logs from all running docker containers

# Description

There was a [discussion on Slack](https://medic.slack.com/archives/C02F2TAME90/p1674141851263569) recently about having a script we could encourage folks to run who were having troubles running CHT 4.x's docker containers. The discussion was in response to [a forum post](https://forum.communityhealthtoolkit.org/t/error-502-when-running-local-cht-instance/2414).  We thought the script would gather up all the logs and create a .tar.gz file that could easily be posted.  This PR adds a bash script to do just that. 

To test:
* create a CHT 4.x instance using[ docker helper](https://docs.communityhealthtoolkit.org/apps/guides/hosting/4.x/app-developer/#cht-docker-helper-for-4x)
* clone the `cht-core` repo, and check out the `add-script-to-zip-docker-logs` branch
* Change directories into the `cht-core` repo: `cd cht-core`
* run the script `./scripts/compress_and_archive_docker_logs.sh`
* check the `support_logs` directory and ensure there's a zip file in there: `ls ~/.medic/support_logs/`
* uncompressing the tarball should yield one log file per container with the container name as the file name. Also a file for `docker ps` and `docker stats` should be saved as `docker_ps.log` and `docker_stats.log`respectively.
* check that the `*log` files have correct contents

# Code review checklist


# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:add-script-to-zip-docker-logs/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:add-script-to-zip-docker-logs/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:add-script-to-zip-docker-logs/docker-compose/cht-couchdb-clustered.yml)
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.
